bin/fin

```diff
@@ -4711,66 +4719,75 @@ _exec ()
 	fi
 
 	# Allow overriding container, where to run
+	local container_name
 	if [[ "$1" =~ "--in" ]]; then
-		CONTAINER_NAME=$(echo "$1" | cut -d "=" -f 2)
+		container_name=$(echo "$1" | cut -d "=" -f 2)
 		shift
+	else
+		container_name="cli"
 	fi
 
-	CONTAINER_NAME=${CONTAINER_NAME:-cli}
-	container_id=$(get_container_id "$CONTAINER_NAME")
+	container_id=$(get_container_id "$container_name")
+
+	# Use sh as default shell
+	local SHELL_INTERACTIVE="sh -ilc"
+	local SHELL_NONINTERACTIVE="sh -lc"
+
+	# For cli use bash
+	if [[ "$container_name" == "cli" ]] || [[ "$container_name" == "db" ]]; then
+		SHELL_INTERACTIVE="bash -ilc"
+		SHELL_NONINTERACTIVE="bash -lc"
+	fi
 
 	# ------------------------------------------------ #
 	# 1) cmd
-	local cmd
+	local cmd=""
 
-	local cdir
 	# Only chdir to the same dir in cli container
 	# RUN_NO_CDIR can be used to override this (used in mysql_import)
-	if [[ "$CONTAINER_NAME" == "cli" ]] && [[ "$RUN_NO_CDIR" != 1 ]]; then
+	if [[ "$container_name" == "cli" ]] && [[ "$RUN_NO_CDIR" != 1 ]]; then
 		local path=$(get_current_relative_path)
 		if [[ "$path" != "" ]] ; then
 			# We are deeper than project root and thus need to do a cd
-			cdir="cd $path &&"
+			cmd="cd $path &&"
 		fi
 	fi
-
-	cmd="$cdir"
 	# ------------------------------------------------ #
 
 	# ------------------------------------------------ #
 	# 2) convert array of parameters into escaped string
+	# Add space if cmd is not empty
+	[[ $cmd != "" ]] && cmd="$cmd "
 	# Escape spaces that are "spaces" and not parameter delimiters (i.e., param1 param2\ with\ spaces param3)
 	if [[ $2 != "" ]]; then
-		cmd="$cmd "$(printf " %q" "$@")
+		cmd="$cmd"$(printf " %q" "$@")
 	# Do not escape spaces if there is only one parameter (e.g., fin run "ls -la | grep txt")
 	else
-		cmd="$cmd $@"
+		cmd="$cmd""$@"
```

Comment:
`cmd="${cmd}${@}"` would be a cleaner option

Reply:
done